keps/sig-storage/2639-secret-configmap-protection/README.md

```diff
@@ -0,0 +1,652 @@
+  - Describe the mechanism:
+    - Prerequisite: InUseProtection feature gate is enabled.
+    - To enable:
+      - Deploy secret-protection-controller to enable Secret protection
+      - Deploy configmap-protection-controller to enable Configmap protection
+    - To disable:
+      - Undeploy secret-protection-controller to disable Secret protection
+      - Undeploy configmap-protection-controller to disable Configmap protection
+  - Will enabling / disabling the feature require downtime of the control
+    plane? No.
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled). No.
+
+###### Does enabling the feature change any default behavior?
+
+Yes. Secrets/ConfigMaps aren't deleted until the resources using them are deleted.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. By undeploying the controllers.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Secrets/ConfigMaps aren't deleted until the resources using them are deleted, again.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes.
+
+- Unit tests cover scenarios for lien feature (InUseProtection feature gate) disabled case.
+- E2e tests for secret-protection-controller and configmap-protection-controller cover scenarios where the controller is deployed and undeployed.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+-->
+
+###### What specific metrics should inform a rollback?
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can an operator determine if the feature is in use by workloads?
+
+There will be Secrets/ConfigMaps which have liens prefixed with `kubernetes.io/secret-protection` or `kubernetes.io/configmap-protection`.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [x] Metrics
+  - Metric name: secret_protection_controller
```

Comment:
I guess it's a typo - I think that metric names will be different (it's the controller), right?

[We don't have to fill this in for Alpha though.]